ClinGen/clincoded
# Connects to #494. Connects to #519. Delete function expansion

Testing:

1. Create GDM
2. Create all sorts of different Group, Family, Individual, Experimental Data combos (w/ children, w/o children, different parents, etc)
3. Try to delete them. Confirm that modal display is correct, and deletions occur properly.
  * Do above as both Admin and Curator
  * Confirm Delete works on different browsers
4. Confirm that the history items properly reflect delete operations
5. If logged in as Admin, confirm that you can still see deleted items in search/indexes
6. If logged in as Curator, confirm that you CANNOT see deleted items in search/indexes
